From f26e0b6fc66528db9b61cb5e9e5e832b210b053e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 1 Mar 2016 19:55:41 +0000
Subject: [PATCH 0738/1077] MANTIS 0037106 bugfix: Trace.dll: call
 Config.removeListener on closeTrace

trace.dll, tracehdl.dll, traceNoConfig.dll, tracehdlNoConfig.dll
V1.0.0.8

- cleanup configuration on close
- removed second call of getIndex (for tracing)



git-svn-id: svn://localhost/SelfServiceCommon/trunk@996 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Trace/inc/TraceConfig.h |  1 +
 SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h   | 12 +++++++++---
 .../Massai/cpp/Trace/src/TraceConfig.cpp             |  9 ++++++++-
 .../Massai/cpp/Trace/src/TraceProcessor.cpp          |  4 +++-
 .../Massai/cpp/Trace/src/TraceSession.cpp            |  1 +
 .../Massai/cpp/Trace/src/TraceSrvConfig.cpp          | 10 ++++++----
 6 files changed, 28 insertions(+), 9 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Trace/inc/TraceConfig.h b/SelfServiceCommon/Massai/cpp/Trace/inc/TraceConfig.h
index 95e4aaf7..ca61ae63 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/inc/TraceConfig.h
+++ b/SelfServiceCommon/Massai/cpp/Trace/inc/TraceConfig.h
@@ -80,6 +80,7 @@ namespace MASSAI
         virtual ~TraceConfig();
 
         void init();
+        void finit();
 
         const std::string& getSessionId() const;
         void setSessionId(const std::string& sessionId);
diff --git a/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h b/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
index f59af714..1b8e7e09 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
+++ b/SelfServiceCommon/Massai/cpp/Trace/inc/local_def.h
@@ -1,5 +1,5 @@
-#define FILE_VER_BIN     1,0,0,7
-#define FILE_VER_STR     "FileVersion", "1.0.0.7"
+#define FILE_VER_BIN     1,0,0,8
+#define FILE_VER_STR     "FileVersion", "1.0.0.8"
 
 /*!
 \file
@@ -9,10 +9,16 @@
 
 \section changehistory Change history
 
-\version 1.0.0.7
+\version 1.0.0.8
 
 trace.dll, tracehdl.dll, traceNoConfig.dll, tracehdlNoConfig.dll
 
+\version 1.0.0.8
+\li 01.03.2016
+\li bugfix SteT - Mantis 0037106: Trace.dll: call Config.removeListener on closeTrace
+- cleanup configuration on close
+- removed second call of getIndex (for tracing)
+
 \version 1.0.0.7
 \li 22.05.2015
 \li bugfix SteT - Mantis 0035476: Trace: wrong timestamp on "start new file"
diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceConfig.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceConfig.cpp
index c1aab256..cab3b50f 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceConfig.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceConfig.cpp
@@ -125,7 +125,9 @@ namespace MASSAI
     }
 
     void TraceCommonConfig::stop()
-    {}
+    {
+      m_pSrvCommonConfig.reset();
+    }
 
     TraceCommonConfig::~TraceCommonConfig()
     {}
@@ -186,6 +188,11 @@ namespace MASSAI
       }
     }
 
+    void TraceConfig::finit()
+    {
+      m_pSrvConfig.reset();
+    }
+
     void TraceConfig::setSessionId(const std::string& sessionId)
     {
       m_sessionId= sessionId;
diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceProcessor.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceProcessor.cpp
index 4012a089..984e97b4 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceProcessor.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceProcessor.cpp
@@ -39,7 +39,9 @@ namespace MASSAI
     }
 
     void TraceProcessor::stop()
-    {}
+    {
+      m_config.stop();
+    }
 
     void TraceProcessor::setThreadIndex(TraceCmd& cmd)
     {
diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceSession.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceSession.cpp
index 8983e21d..2d9b73fb 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceSession.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceSession.cpp
@@ -70,6 +70,7 @@ namespace MASSAI
     void TraceSession::finit()
     {
       m_init = false;
+      m_config.finit();
     }
 
     void TraceSession::open()
diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceSrvConfig.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceSrvConfig.cpp
index 552e119a..3919e11f 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceSrvConfig.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceSrvConfig.cpp
@@ -160,8 +160,9 @@ namespace MASSAI
 
     long TraceLevel::getLevel()
     {
-      tracetrace(BasicRuntime_HERE, "level: (%.100s)(%d)", item.c_str(), getIndex());
-      return getIndex();
+      long result = getIndex();
+      tracetrace(BasicRuntime_HERE, "level: (%.100s)(%d)", item.c_str(), result);
+      return result;
     }
 
     void TraceLevel::create(char const*& Rights, char const*& Explanation, long& DefaultIndex, nConfig::selection const*& Begin, nConfig::selection const*& End)
@@ -190,8 +191,9 @@ namespace MASSAI
 
     bool TraceTopic::isActive()
     {
-      tracetrace(BasicRuntime_HERE, "topic: (%.100s)(%d)", item.c_str(), getIndex());
-      return getIndex() == TS_ACTIVE;
+      long index = getIndex();
+      tracetrace(BasicRuntime_HERE, "topic: (%.100s)(%d)", item.c_str(), index);
+      return index == TS_ACTIVE;
     }
 
     void TraceTopic::create(char const*& Rights, char const*& Explanation, long& DefaultIndex, nConfig::selection const*& Begin, nConfig::selection const*& End)
-- 
2.41.0.windows.1

